docs/oci-storage.md

```diff
@@ -1,40 +1,152 @@
+  * Fleet allows these flags for development and testing purposes, but they should never be used in production.  
+  * If you use these tags, you expose your app to security vulnerabilities, and put cluster workloads, and credentials at a risk of tampering with.  
+* Changing the secret does not update the deployments, only on the next GIT update it would use the new storage registry.
+
+
+Fleet uses `gitcredential` secret when the `clientSecretName` field is empty in a GitRepo. It uses global `ocistorage` as a fallback if `ociRegistrySecret` is not defined.
+
+:::note
+Fleet checks for the integrity of OCI artifacts and Fleet tags OCI artifact as Latest
+:::
+
+### Fleet Example
+
+Consider the GitRepo file for fleet-examples repository. 
+
+```yaml
+apiVersion: fleet.cattle.io/v1alpha1
+kind: GitRepo
+metadata:
+ name: frontend-oci
+ namespace: fleet-local
+spec:
+ repo: https://github.com/your-org/fleet-oci-example.git
+ branch: main
+ paths:
+ - ./frontend/frontend-deployment.yaml
+    - ./frontend/frontend-service.yaml
+  ociRegistrySecret: ocistorage
+```
+
+To define OCI Secret, you must create an oci-secret.yaml file
 
 ```yaml
-// when ociRegistry is defined Fleet will use oci registry as storage
-ociRegistry:
-    // reference is the OCI registry url.
-    reference: "docker.io/your-user-here"
-    // secret name where the credentials for the OCI registry are.
-    // expects a generic secret with username and password keys set.
-    authSecretName: oci-secret
-    // basicHTTP allows Fleet to use basic http connections to communicate
-    // with the registry (defaults to false)
-    basicHTTP: false
-    // insecureSkipTLS allows connections to the OCI registry
-    // without certs (defaults to false)
-    insecureSkipTLS: false
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ocistorage
+  namespace: fleet-local
+type: fleet.cattle.io/bundle-oci-storage/v1alpha1
+data:
+  username: <base64-encoded-user>
+  password: <base64-encoded-password>
+  insecureSkipTLS: <base64-encoded-true/false>
+  basicHTTP: <base64-encoded-true/false>
+  agentUsername: <base64-encoded-readonly-user>
+  agentPassword: <base64-encoded-password>
+  
 ```
+
+Run `kubectl apply -f secrets/oci-secret.yaml` before applying the GitRepo.
+
+Or you can use `kubectl` command to create the `ocistorage` secret using unencoded text. Kubernetes converts them to base64 encoded for storing the secret.
+
+```bash
+kubectl -n fleet-local create secret generic ocistorage \
+  --type=fleet.cattle.io/bundle-oci-storage/v1alpha1 \
+  --from-literal=username=fleet-ci \
+  --from-literal=password=fleetRocks \
+  --from-literal=reference=192.168.1.39:8082 \
+  --from-literal=insecureSkipTLS=true \
+  --from-literal=basicHTTP=false \
+  --from-literal=agentUsername=fleet-ci-readonly \
+  --from-literal=agentPassword=readonlypass
+```
+
+To validate your secret, you can run:
+
+`kubectl get secret ocistorage -n fleet-local -o yaml`
```

Comment:
I would also add the command to show the values of the secret decoded.
In fact I see you list the command in the screenshot.

Validating the secret is, anyway, part of kubernetes itself, it's not tied to Fleet.